docs/rules/0158/response-repeated-first-field.md

```diff
@@ -0,0 +1,65 @@
+---
+rule:
+  aip: 158
+  name: [core, '0158', response-repeated-first-field]
+  summary:
+    First field (by both position and field number) of Paginated RPCs' response
+    should be repeated.
+permalink: /158/response-repeated-first-field
+redirect_from:
+  - /0158/response-repeated-first-field
+---
+
+# Paginated methods: Page token field
+
+This rule enforces that all `List` and `Search` methods have a repeatable field
+as a first field in the response message, as mandated in [AIP-158][].
+
+## Details
+
+This rule looks at any message matching `List*Response` or `Search*Response`
+that has `next_page_token` field and complains if the first field (by both
```

Comment:
Why does it matter if they have a `next_page_token` field? For example, shouldn't we warn on the following:

```proto
message ListBooks {
  Book book = 1;
}
```

and say that the correct version is

```proto
message ListBooks {
  repeated Book books = 1;
}
```

?

Reply:
Maybe make it a disjunction? I definitely want to match any response message that has `string next_page_token` even if it does not have a message name I recognize. It seems reasonable that I also want to match any ListFooResponse or SearchFooResponse

Reply:
Discussed in this PR: https://github.com/googleapis/api-linter/pull/496